### DISCUSSION TITLE
Hardware with button ?

### DISCUSSION THREAD

**Discussion by mr-sneezy** [rbaron/b-parasite#93] [General]
I heard from guys on Home Assistant forums that there is or was a HW revision with a button onboard for easier Zigbee binding planned.
I check the HW-v2 gerbers but no button in that schematic. 
Anyone know if the button is still planned on the HW ?
My reason is I'd like to get some boards made for myself but don't want make boards that will shortly be obsoleted...

**Comment by rbaron** [comment]
There are two buttons in the `hardware-v2` branch (RST and SW1) - I just haven't re-rendered the PDF from the kicad_sch there yet. I will hopefully have some free time soon to review, order a test batch and merge it if all goes well.

That being said, the buttonless designs (1.0.0, 1.1.0, 1.2.0) work fine with the main BLE and seem to be working okay with the experimental Zigbee sample firmware. I managed to implement two factory reset methods that don't require physical buttons, the double-reset being the default one. This makes me question the usefulness of physical buttons, as moving parts near water are a sort of a liability at arguably no extra functionality. I will probably still go for it at least for a trial run, but I quite like the simplicity and reliability of the buttonless design, and I will keep support for them (I myself still have more than a few 1.0.0 boards from two years ago running the latest firmware).

**Comment by mr-sneezy** [comment]
As long as binding is easy for somebody with relatively poor eyesight it's all good, shorting tiny RST pads to ground is something I avoid due to mishaps (though a SMD micro momentary is sometimes not much easier).
I will likely make a protective case in the form of a one piece cap that snugs onto a base so there are no seams on the sides anyway, re the possible water issue.
Looking forward to the update, my HA watering system is not really very intelligent until it gets some soil moisture feedback in place.

**Comment by rbaron** [comment]
Side note: as it stands the Zigbee sample first boots into pairing mode. You'd need to double reset only to factory reset it if you ever need it, and it can also be done by briefly removing/inserting the battery twice (_or_ shorting RST to ground). 100% agree that a push button is a better UI.

**Comment by rbaron** [comment]
Closing - we launched 2.0.0 with a push button.
